storage/client/client.go

```diff
@@ -1437,13 +1451,63 @@ func (c *StorageClient) RuntimeEvents(ctx context.Context, p apiTypes.GetRuntime
 	return &es, nil
 }
 
+// Fetch account balances from the node and pass them back via the channel. Note that if the input context
+// times out or we encounter an error, we explicitly close the channel and return. This enables the calling
+// function to create a context with a timeout and then listen to the channel with the expectation that the
+// channel will return values and/or close approximately within the timeout.
+func (c *StorageClient) fetchAccountBalanceFromNode(ctx context.Context, ch chan *RuntimeSdkBalance, runtime common.Runtime, address staking.Address) {
+	runtimeApi, ok := c.runtimeClients[runtime]
+	if !ok {
+		c.logger.Warn("no runtime api configured to fetch account balances", "runtime", runtime)
+		close(ch)
+		return
+	}
+	balances, err := runtimeApi.GetBalances(ctx, roothash.RoundLatest, address)
+	if err != nil {
+		c.logger.Warn("failed to fetch balances from node", "address", address.String(), "runtime", runtime, "round", roothash.RoundLatest, "err", err)
+		close(ch)
+		return
+	}
+	foundNativeBalance := false
+	for denom, amount := range balances {
+		balance := RuntimeSdkBalance{
+			Balance: amount,
+			// HACK: 18 is accurate for Emerald and Sapphire, but Cipher has 9.
+			// Once we add a non-18-decimals runtime, we'll need to query the runtime for this
+			// at analysis time and store it in a table, similar to how we store the EVM token metadata.
+			TokenDecimals: 18,
+			TokenSymbol:   denom.String(),
+		}
+		if denom.IsNative() {
+			foundNativeBalance = true
```

Comment:
imo send the whole balances map back, we might as well offer all this latest data in the api

Reply:
Discussed in product sync; decision was to only query at the latest indexed height and have the explorer just warn the user if it's falling behind.